subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/GradleBuildInstantExecutionSmokeTest.groovy

```diff
@@ -41,27 +54,44 @@ class GradleBuildInstantExecutionSmokeTest extends AbstractSmokeTest {
         result.output.count("Calculating task graph as no instant execution cache is available") == 1
 
         when:
+        run("clean")
+
+        and:
         result = instantRun(*supportedTasks)
 
         then:
         result.output.count("Reusing instant execution cache") == 1
```

Comment:
perhaps verify that the test results are available and the test was executed (using the `DefaultTestExecutionResult` test fixtures to do the check).